src/api/options.c

```diff
@@ -654,7 +659,7 @@ nlopt_result NLOPT_STDCALL nlopt_set_x_weights(nlopt_opt opt, const double *x_we
 
 nlopt_result NLOPT_STDCALL nlopt_set_x_weights1(nlopt_opt opt, double x_weights)
 {
-    if (opt) {
+    if (opt && x_weights > 0) {
```

Comment:
Better to have
```c
if (x_weights < 0) return ERR(NLOPT_INVALID_ARGS, opt, "invalid negative weight");
```
so that there is an informative error message.

Reply:
Should we make `x_weights == 0` an error, too? I am assuming we should, since there already is `xtol_rel = 0` to disable the stopping criterion.